x/profiles/types/chain_link.go

```diff
@@ -0,0 +1,158 @@
+package types
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/ghodss/yaml"
+)
+
+// NewChainConfig is a constructor function for ChainConfig
+func NewChainConfig(name string, prefix string) ChainConfig {
+	return ChainConfig{
+		Name:             name,
+		Bech32AddrPrefix: prefix,
+	}
+}
+
+func (chainConfig ChainConfig) Validate() error {
+	if chainConfig.Name == "" {
+		return fmt.Errorf("chain name cannot be empty")
+	}
+	return nil
```

Comment:
We should also check to make sure the Bech32 prefix is not empty

Reply:
It might be empty if the link is from the chain without bech32 encoded address.

Reply:
@dadamu Are there chains that do not use the Bech 32 encoding for their addresses?

Reply:
@RiccardoM Solana is using base58 public key as address.

Reply:
@dadamu Ok, I see. Than instead of 

```proto
message ChainLink {
  string address = 1;
  // ...
}
```

I think it might be better to use something like 
```proto
message ChainLink {
  Address address = 1; 
  // ...
}

message Address {
  oneof sum {
    Bech32 bech_32 = 1;
    Base58 base_58 = 2;

  }

  message Bech32 {
    string value = 1;
    string prefix = 2;
  }

  message Base58 {
    string value = 1;
  }
}
```

This way we will be able to represent various encoding methods differently with each one their additional data (eg. the `prefix` for Bech32). If this approach is used, we can then remove the `bech32_addr_prefix` from the `ChainConfig` object.

Reply:
I stuck on the RegisterInterfaces in `codec.go` when implementing.

When it tried to `GetSignBytes()` in link chain account message, there is a panic said `"panic: Unregistered interface types.isAddress_Sum"`.
Then I tried to register `Address_Bech_32` and `Address_Base_58` type to `codec` like:
```
registry.RegisterImplementations((*isAddress_Sum)(nil), &Address_Bech_32{}, &Address_Base_58{})
```
However, It shows the message that `cannot use &(Address_Base_58 literal) (value of type *Address_Base_58) as proto.Message value in argument to registry.RegisterImplementations: missing method ProtoMessage`.

Could you help me? @RiccardoM @bragaz

Reply:
I have an idea that is using the following struct to avoid the problem I mentioned above.
```proto
message MsgLinkAccount{
    string source_address = 1;
    string source_prefix =2;
    ...
}
```
What do you think?

Reply:
@RiccardoM I got it, `Message` and `Packet` should also use the `Any` type on `source_address` field. Not sure if it occurs the marshaling error.  I would try it first.

Reply:
@RiccardoM I got the problem that when using msg `GetSigners()` method, it should have unpacker to unpack the `Any` type address.

Could we have a simple `AddressData` type shows if it has prefix or not? Say:
```proto
message AddressData{
  string address
  string prefix
}
```
If the chain address doesn't have prefix, just let it empty string.

Reply:
@RiccardoM I think a while about the `MsgLinkChainAccount` should have a `signer` field for `GetSigners()` function. In the case that the source account is from `Solana`. The msg `GetSigners()` will get the `Solana` address.

```proto
message MsgLinkChainAccount{
  string signer = 1;
  types.Any source_address = 2 ; // implement AddressData interface
  // ...
}
```

What do you think?

Reply:
@dadamu Ideally, the message signer is going to be the destination address. This will represent the address associated to the Desmos profile, and for this reason will always be a valid Desmos address.

When I link an external chain to my profile, I should be the message signer. There should not be another signer aside from me